Tools/WinMLRunner/src/OutputHelper.h

```diff
@@ -667,24 +688,108 @@ class OutputHelper
     }
 
     template<typename T>
-    void WriteTensorResultToCSV(T* m_Res, uint32_t iterationNum, const CommandLineArgs &args, uint32_t &capacity, std::string &featureName)
+    void ProcessTensorResult(com_ptr<ITensorNative> itn,
```

Comment:
Try to detemplatize so there isn't code repetition. You can just use a switch statement in the innermost loop.

Reply:
Removed out most of the code that's unnecessary for being inside the template. Case switch statement for tensorkind will have to exist somewhere. I chose to have it in the helper method that calls ProcessTensorResult and leave ProcessTensorResult as template for using typename T for looping through the buffer.